### PR TITLE
Issue/ldap Config profile fix

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

Modify LDAP Config:

- Added profile 
- Removed API keys
- fixed readonly properties

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/ldap-configuration/cmd/resource/resource.go [hunk 1]
```diff
@@ -36,12 +38,12 @@ const (
...
-var CreateRequiredFields = []string{constants.PubKey, constants.PvtKey, constants.GroupID,
+var CreateRequiredFields = []string{constants.GroupID,
 	BindUsername, BindPassword, Hostname, Port}
-var ReadRequiredFields = []string{constants.GroupID, constants.PubKey, constants.PvtKey}
-var UpdateRequiredFields = []string{constants.GroupID, constants.PubKey, constants.PvtKey}
-var DeleteRequiredFields = []string{constants.GroupID, constants.PubKey, constants.PvtKey}
-var ListRequiredFields = []string{constants.PubKey, constants.PvtKey}
+var ReadRequiredFields = []string{constants.GroupID}
+var UpdateRequiredFields = []string{constants.GroupID}
+var DeleteRequiredFields = []string{constants.GroupID}
```
Comment (colm-quinn) [inline cfn-resources/ldap-configuration/cmd/resource/resource.go:45]: From the other PRs, we've been standardising on ProjectID over GroupID. Good to be consistent, especially as these changes would be a breaking change in the future.

File: cfn-resources/ldap-configuration/cmd/resource/resource.go [hunk 2]
```diff
@@ -118,10 +120,14 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *modelValidation, nil
 	}
 
+	if currentModel.Profile == nil {
+		currentModel.Profile = pointy.String(userprofile.DefaultProfile)
+	}
```
Comment (colm-quinn) [inline cfn-resources/ldap-configuration/cmd/resource/resource.go:125]:
```suggestion
	if currentModel.Profile == nil || *currentModel.Profile == "" {
		currentModel.Profile = aws.String(profile.DefaultProfile)
	}
```

Comment (colm-quinn) [reply cfn-resources/ldap-configuration/cmd/resource/resource.go:125]: Also moving from pointy to aws.string for consistency

File: cfn-resources/ldap-configuration/cmd/resource/resource.go [hunk 3]
```diff
@@ -156,11 +162,14 @@ func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 		return *modelValidation, nil
 	}
 
+	if currentModel.Profile == nil {
```
Comment (colm-quinn) [inline cfn-resources/ldap-configuration/cmd/resource/resource.go:165]: Same as above suggestion

File: cfn-resources/ldap-configuration/docs/README.md [hunk 4]
```diff
@@ -12,15 +12,16 @@ To declare this entity in your AWS CloudFormation template, use the following sy
...
         "<a href="#bindusername" title="BindUsername">BindUsername</a>" : <i>String</i>,
         "<a href="#status" title="Status">Status</a>" : <i>String</i>,
         "<a href="#hostname" title="Hostname">Hostname</a>" : <i>String</i>,
-        "<a href="#apikeys" title="ApiKeys">ApiKeys</a>" : <i><a href="apikeydefinition.md">ApiKeyDefinition</a></i>,
         "<a href="#authenticationenabled" title="AuthenticationEnabled">AuthenticationEnabled</a>" : <i>Boolean</i>,
         "<a href="#authorizationenabled" title="AuthorizationEnabled">AuthorizationEnabled</a>" : <i>Boolean</i>,
         "<a href="#cacertificate" title="CaCertificate">CaCertificate</a>" : <i>String</i>,
         "<a href="#authzquerytemplate" title="AuthzQueryTemplate">AuthzQueryTemplate</a>" : <i>String</i>,
         "<a href="#bindpassword" title="BindPassword">BindPassword</a>" : <i>String</i>,
+        "<a href="#groupid" title="GroupId">GroupId</a>" : <i>String</i>,
```
Comment (colm-quinn) [inline cfn-resources/ldap-configuration/docs/README.md:24]:
```suggestion
        "<a href="#groupid" title="ProjectId">ProjectId</a>" : <i>String</i>,
```

Same in other other occurances

File: cfn-resources/ldap-configuration/docs/README.md [hunk 5]
```diff
@@ -32,22 +33,33 @@ To declare this entity in your AWS CloudFormation template, use the following sy
...
     <a href="#port" title="Port">Port</a>: <i>Integer</i>
     <a href="#usertodnmapping" title="UserToDNMapping">UserToDNMapping</a>: <i>
       - <a href="apiatlasndsusertodnmappingview.md">ApiAtlasNDSUserToDNMappingView</a></i>
 </pre>
 
 ## Properties
 
+#### Profile
+
+Profile used to provide credentials information, (a secret with the cfn/atlas/profile/{Profile}, is required), if not provided default is used
```
Comment (colm-quinn) [inline cfn-resources/ldap-configuration/docs/README.md:55]:
```suggestion
The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).
```

**Review by colm-quinn — COMMENTED** [review]
Looking good, just consistency updates with the other resource changes we've made to client, projected & profile help text, then LGTM

File: cfn-resources/ldap-configuration/mongodb-atlas-ldapconfiguration.json [hunk 6]
```diff
@@ -50,6 +38,11 @@
     }
   },
   "properties": {
+    "Profile": {
+      "type": "string",
+      "description": "Profile used to provide credentials information, (a secret with the cfn/atlas/profile/{Profile}, is required), if not provided default is used",
```
[the remark this answers is not in the document]
Comment (Adelmar92) [reply cfn-resources/ldap-configuration/mongodb-atlas-ldapconfiguration.json:43]: i would use a link here, this description is used in CloudFormation, and the link will be broken, and will render in a weird way

**Review by colm-quinn — DISMISSED** [review]
Thank you for all the changes, LGTM!

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**